src/plugins/mer/meroptionspage.cpp

```diff
@@ -66,7 +66,8 @@ void MerOptionsPage::apply()
 
 void MerOptionsPage::finish()
 {
-    delete m_widget;
+    if(m_widget)
+        m_widget->close();
```

Comment:
I checked implementation of other options pages, most of them do delete m_widget in their finish(). To me it is a sign this is not fixing the actual source of the bug. Could you please recheck?

Reply:
Hm. Other way to fix problem use `loop->exec(QEventLoop::ExcludeUserInputEvents);` in MerConnection::vmPollState(Synchronous) to really by a synchronous func.

Reply:
OK, now I see where the issues comes from. There are changes in progress that are likely to influence this - let's wait with this PR and recheck later after that changes are completed.